### ISSUE TITLE
[TEST][TENSORFLOW] Cache the Downloaded File

### ISSUE THREAD

**Issue by tqchen** [dmlc/tvm#2911]
So far the model files used in the TF end to end tests re-download the file in each test run. 
This causes the test execution to be slow. Eventually, the test server can be blocked by the place that hosts the data-source.

We need to change the implementation to cache to local and only re-download the file if necessary.

cc @srkreddy1238 @icemelon9

**Comment by tqchen** [comment]
http://ci.tvm.ai:8080/blue/rest/organizations/jenkins/pipelines/tvm/branches/master/runs/896/nodes/236/log/?start=0

**Comment by tqchen** [comment]
A quick possible solution: instead of download, have a util to do a lazy git clone and checkout of the data repo and get the file there

**Comment by srkreddy1238** [comment]
I will send a PR soon on this

**Comment by tqchen** [comment]
cc @icemelon9 said he is looking into it, perhaps you can coordinate

**Comment by icemelon9** [comment]
@srkreddy1238 I'm looking into this issue now. If you've already done something, go ahead and just let me know. Otherwise, I can work on it.

**Comment by srkreddy1238** [comment]
Great. Please proceed. I will help in review
